src/main/java/com/microsoft/sqlserver/jdbc/SQLServerAeadAes256CbcHmac256Algorithm.java

```diff
@@ -290,31 +298,30 @@
      */
     private byte[] decryptData(byte[] iv, byte[] cipherText, int offset, int count) throws SQLServerException {
         aeLogger.entering(SQLServerAeadAes256CbcHmac256Algorithm.class.getName(),
-                Thread.currentThread().getStackTrace()[1].getMethodName(), "Decrypting data.");
+                "decryptData", "Decrypting data.");
         assert (cipherText != null);
         assert (iv != null);
         byte[] plainText = null;
         // key to be used for decryption
         SecretKeySpec skeySpec = new SecretKeySpec(columnEncryptionkey.getEncryptionKey(), "AES");
         IvParameterSpec ivector = new IvParameterSpec(iv);
-        Cipher decryptCipher;
         try {
             // AES encryption CBC mode and PKCS5 padding
-            decryptCipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
-            decryptCipher.init(Cipher.DECRYPT_MODE, skeySpec, ivector);
+            if (decryptCipher != null) {

```

Comment:
Why not instantiate and initialize the decryptCipher here itself? Why is a static member to this class needed when the `Cipher.getInstance(...)` itself is static?

Reply:
Looks like the code earlier was doing what I am talking about, above. Why is this change for declaring `decryptCipher` as a static class member needed?

Reply:
There is no reason for the decryptCipher variable to be static, I will change this. However, I believe it should be done once in the constructor as there isn't a point to get an instance for every decryptData(byte,byte,int,int) call.

Reply:
The issue I see with having this in the constructor is that you are replicating the behavior of Cipher.getInstance() which is static itself and adding this additional layer of caching and making assumption about the API that it would be safe to use the same object when the API itself doesn't make any such guarantees in the documentation. 
A static is good if you want to cache the object at your end and it is safe to do so, but since Cipher.getInstance(...) is static itself, it should decide whether it should give a new object everytime or return a cached object. 

In case the behavior of Cipher.getInstance() changes for any reason across JDK versions, then this driver will have to comb through this caching. Such bugs are hard to detect. Such layers of caching of objects returned from static call is not a good practice.

Reply:
Agreed, I will revert the changes.